src/components/side_panel/table_panel/table_panel.ts

```diff
@@ -53,23 +53,33 @@ export class TablePanel extends Component<Props, SpreadsheetChildEnv> {
   static props = { onCloseSidePanel: Function, table: Object };
 
   state!: State;
+  /** We want to save the state of the "hasFilter" checkbox so that toggling the header off (which
+   * disable the filters), then toggling it back on doesn't change the "hasFilter" state*/
+  filtersEnabledIfPossible: boolean = false;
```

Comment:
could/should be in the (reactive) state. It's error prone and here's no reason not to move it there

Reply:
But won't it trigger another render when we update it in the onWillUpdateProps then ?